native/cocos/base/threading/MessageQueue.cpp

```diff
@@ -76,6 +76,16 @@ void MessageQueue::MemoryAllocator::freeByUser(MessageQueue *const mainMessageQu
     mainMessageQueue->kick();
 }
 
+ MessageQueue::MemoryAllocator::~MemoryAllocator() noexcept {
+    uint8_t *chunk = nullptr;
+     while (_chunkCount.load(std::memory_order_acquire) != 0) {
+         if(_chunkPool.try_dequeue(chunk)) {
+             free(chunk);
+             _chunkCount.fetch_sub(1, std::memory_order_acq_rel);
+         }
+     }
```

Comment:
I think only need load one time, and free all chunks, as it is a destructor, so can modify like
```c++
uint32_t chunkCount = _chunkCount.load(std::memory_order_acquire);
while (chunkCount--) {
    if(_chunkPool.try_dequeue(chunk)) {
             free(chunk);
         }
} 
```

Reply:
And i think can even change to
```c++
while (_chunkPool.try_dequeue(chunk)) {
             free(chunk);
} 
```